### PR TITLE
⚡ Activities for addressbook and contacts

### LINKED ISSUE #3549: Contact changes in activity stream
This was briefly discussed in https://github.com/nextcloud/contacts/issues/141

### Steps to reproduce
1. change a contact
2. not showing in activity stream
3.

### Expected behaviour
changes should show up in activity stream

### Actual behaviour
Not showing up in activity stream

### Server configuration
**Nextcloud version:**  11.0.1

**Contacts version:** 1.5.3

**Updated from an older Nextcloud or fresh install:** updated

As a user I would like to see changes to contacts (especially on shared address books) to show up in activity stream or even trigger an e-mail to me. Other apps (calendar for example) already do this.

Thanks!

From the other discussion:
https://github.com/nextcloud/server/issues/1662 (contacts trashbin)
https://github.com/nextcloud/server/blob/9e32885b34dbf6341551ad981365d5d1742a4f65/apps/dav/lib/CalDAV/Activity/Backend.php#L68 (calendar activity "hooks")

### PR DESCRIPTION
**Pull request by nickvergessen**
- [x] Create addressbook
- [x] Update addressbook
- [x] Delete addressbook
- [x] (un)Share addressbook
- [x] Create contact
- [x] Update contact
- [x] Delete contact

Fix #3549

Bildschirmfoto von 2021-05-21 15-05-38

### CODE REVIEW

File: apps/dav/lib/CardDAV/Activity/Filter.php [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+	protected $url;
+
+	public function __construct(IL10N $l, IURLGenerator $url) {
+		$this->l = $l;
+		$this->url = $url;
+	}
+
+	/**
+	 * @return string Lowercase a-z and underscore only identifier
+	 * @since 11.0.0
```
Comment (skjnldsv) [inline apps/dav/lib/CardDAV/Activity/Filter.php:45]: 22 ?

Comment (nickvergessen) [reply apps/dav/lib/CardDAV/Activity/Filter.php:45]: No, its the interface being copied.
But should maybe drop it completely from the implementations

**Review by skjnldsv — APPROVED** [review]

**Comment by szaimen** [comment]
seems like a few integration tests are failing? or not related?

**Comment by skjnldsv** [comment]
Seems unrelated

**Review by ChristophWurst — APPROVED** [review]

**Comment by ChristophWurst** [comment]
Just wanted to say this is an awesome feature :rocket: Yet again someone added a strange contact to a shared addressbook and I can now finally see who did it :)

**Pull request merged**